### ISSUE TITLE
Unable to ingest data

### ISSUE THREAD

**Issue by akrogager** [diffgram/python-sdk#46]
I recently started using Diffgram and would like to use the SDK to automatically ingest data. However, the SDK doesn't seem to be working for me. If I run `project.file.from_url(signed_url)`, I get the following error:
```
{'directory': "'directory_id' not supplied"}
```
I tried manually adding the directory id to the packet in the request payload as follows and then the request went through without any errors:
```
response = project.session.post(
    project.host + endpoint,
    json = {**packet, "directory_id": str(project.default_directory.id)})
```
However, I still don't see any file uploaded in the Diffgram app. Any input on how to resolve this would be greatly appreciated.

**Comment by akrogager** [comment]
The problem was that my signed_url expired too fast 🤦 But it would still be great if you could add the directory_id to the from_url method.

**Comment by JKYang01** [comment]
Hi akrogager,
Thank you very much for the information! I am also tying to ingest data using SDK and I encountered exact the same issue like yours. there ara two things I figured out:
1.  signed_url expiration:   If you use AWS to generate the signed url, AWS only allow you set up time out up to 604800 seconds. 
2. the directory_id no supplied error :  the function from_url() did not get the directory_id which was required in the function self.__build_packet_payload() 
my solution is add in the following code under from_url() function in diffgram/file/fileconstructor.py  under line 255

```
if self.client.default_directory is None:
            raise Exception("Directory not set. call set_default_directory() to set upload directory.")
        directory_id = self.client.directory_id
```
after this modification, the pictures are uploaded successfully to the folder
